client/src/api/client.ts

```diff
@@ -0,0 +1,555 @@
+import { errors } from '../util/errors';
+import type * as Types from './components';
+import { operationsByTag } from './components';
+import type { FetcherExtraProps, FetchImpl } from './fetcher';
+import { HostProvider, getHostUrl } from './providers';
+import type * as Responses from './responses';
+import type * as Schemas from './schemas';
+
+export interface XataApiClientOptions {
+  fetchImpl: FetchImpl;
+  apiKey: string;
+  host?: HostProvider;
+}
+
+export class XataApiClient {
+  private extraProps: FetcherExtraProps;
+
+  constructor(options: XataApiClientOptions) {
+    const fetchImpl = typeof fetch !== 'undefined' ? fetch : options.fetchImpl;
+    if (!fetchImpl) {
+      throw new Error(errors.noFetchImplementation);
+    }
+
+    const provider = options.host ?? 'production';
+
+    this.extraProps = {
+      apiUrl: getHostUrl(provider, 'main'),
+      workspacesApiUrl: getHostUrl(provider, 'workspaces'),
+      fetchImpl,
+      apiKey: options.apiKey
+    };
+  }
+
+  public get user() {
```

Comment:
What do you think about using plural in these getters? users, workspaces, databases?